staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -165,49 +195,70 @@ func NewNamespaceKeyedIndexerAndReflector(lw ListerWatcher, expectedType interfa
 // "yes".  This enables you to use reflectors to periodically process
 // everything as well as incrementally processing the things that
 // change.
-func NewReflector(lw ListerWatcher, expectedType interface{}, store Store, resyncPeriod time.Duration) *Reflector {
-	return NewNamedReflector(naming.GetNameFromCallsite(internalPackages...), lw, expectedType, store, resyncPeriod)
-}
-
-// NewNamedReflector same as NewReflector, but with a specified name for logging
-func NewNamedReflector(name string, lw ListerWatcher, expectedType interface{}, store Store, resyncPeriod time.Duration) *Reflector {
+func NewReflectorWithOptions(lw ListerWatcher, expectedType interface{}, store Store, options ReflectorOptions) *Reflector {
```

Comment:
I'm wondering if we should change expectedType from interface{} to runtime.Object in these constructors... I don't think any non-runtime.Object value could produce a functional reflector based on https://github.com/kubernetes/kubernetes/blob/3e26e104bdf9d0dc3c4046d6350b93557c67f3f4/staging/src/k8s.io/client-go/tools/cache/reflector.go#L511-L516

what do you think?

Reply:
Let me try it and see what shakes out

Reply:
@liggitt would you also change the preexisting functions `NewNamespaceKeyedIndexerAndReflector`, `NewReflector`, `NewNamedReflector` to use `runtime.Object`? Or keep them as-is (to avoid breakage) and panic (?) if the passed in `expectedType` is not a `runtime.Object`?

Reply:
Or maybe do this change in a separate PR if desired?

Reply:
I would change them all IFF there were no ripples... like I said, I can't see how someone would successfully call them with a non-runtime.Object.

If there were ripples, I'd defer it to another PR.

Reply:
actually, either way, a separate PR is fine